File: Sources/Linq2Acad.SampleCode.VB/SampleCode.Addin.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Autodesk.AutoCAD.ApplicationServices
Imports Autodesk.AutoCAD.DatabaseServices
Imports Autodesk.AutoCAD.EditorInput
Imports Autodesk.AutoCAD.Runtime

Partial Public Class SampleCode
  Implements IExtensionApplication
  Public Sub Initialize() Implements IExtensionApplication.Initialize
  End Sub

  Public Sub Terminate() Implements IExtensionApplication.Terminate
  End Sub

  Private Function GetString(message As String) As String
    If Application.DocumentManager.MdiActiveDocument IsNot Nothing Then
      Dim editor = Application.DocumentManager.MdiActiveDocument.Editor

      Dim result = editor.GetString(If(message.EndsWith(":"), message, (message & Convert.ToString(":"))))

      If result.Status = PromptStatus.OK Then
        Return result.StringResult
      Else
        Return Nothing
      End If
    Else
      Return Nothing
    End If
  End Function

  Private Function GetEntity(message As String) As ObjectId
    If Application.DocumentManager.MdiActiveDocument IsNot Nothing Then
      Dim editor = Application.DocumentManager.MdiActiveDocument.Editor
      Dim result = editor.GetEntity(message & Convert.ToString(":"))

      If result.Status = PromptStatus.OK Then
        Return result.ObjectId
      Else
        Return ObjectId.Null
      End If
    Else
      Return ObjectId.Null
    End If
  End Function

  Private Sub WriteMessage(message As String)
    If Application.DocumentManager.MdiActiveDocument IsNot Nothing Then
      Dim editor = Application.DocumentManager.MdiActiveDocument.Editor
      editor.WriteLine(message)
    End If
  End Sub
End Class

File: Sources/Linq2Acad.SampleCode.CS/SampleCode.Addin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Autodesk.AutoCAD.ApplicationServices.Core;
using Autodesk.AutoCAD.DatabaseServices;
using Autodesk.AutoCAD.EditorInput;
using Autodesk.AutoCAD.Runtime;

namespace Linq2Acad
{
  public partial class SampleCode : IExtensionApplication
  {
    public void Initialize()
    {
    }

    public void Terminate()
    {
    }

    private string GetString(string message)
    {
      if (Application.DocumentManager.MdiActiveDocument != null)
      {
        var editor = Application.DocumentManager.MdiActiveDocument.Editor;

        var result = editor.GetString(message.EndsWith(":") ? message : (message + ":"));

        if (result.Status == PromptStatus.OK)
        {
          return result.StringResult;
        }
        else
        {
          return null;
        }
      }
      else
      {
        return null;
      }
    }

    private ObjectId GetEntity(string message)
    {
      if (Application.DocumentManager.MdiActiveDocument != null)
      {
        var editor = Application.DocumentManager.MdiActiveDocument.Editor;
        var result = editor.GetEntity(message + ":");

        if (result.Status == PromptStatus.OK)
        {
          return result.ObjectId;
        }
        else
        {
          return ObjectId.Null;
        }
      }
      else
      {
        return ObjectId.Null;
      }
    }

    private void WriteMessage(string message)
    {
      if (Application.DocumentManager.MdiActiveDocument != null)
      {
        var editor = Application.DocumentManager.MdiActiveDocument.Editor;
        editor.WriteLine(message);
      }
    }
  }
}
